### ISSUE TITLE
`manual_is_variant_and` should also trigger for `some_option.map(|x| x.is_foo) == Some(true)`

### ISSUE THREAD

**Issue by sdroege** [rust-lang/rust-clippy#14542]
### Summary

See title :)

### Lint Name

manual_is_variant_and

### Reproducer

I tried this code:

```rust
let x = Some(2).map(|x| x % 2 == 0) == Some(true);
```

I expected to see this happen:

Suggestion to use `let x = Some(2).is_some_and(|x| x % 2 == 0);`

Instead, this happened:

Nothing

### Version

```text
rustc 1.86.0 (05f9846f8 2025-03-31)
binary: rustc
commit-hash: 05f9846f893b09a1be1fc8560e33fc3c815cfecb
commit-date: 2025-03-31
host: x86_64-unknown-linux-gnu
release: 1.86.0
LLVM version: 19.1.7
```

**Comment by GuillaumeGomez** [comment]
Any equality comparison expression where an arm ends with a `map` call returning a bool should be using `is_some_and`. Gonna take a look in the next days.
